Marlin/Marlin_main.cpp

```diff
@@ -5318,6 +5344,21 @@ void home_all_axes() { gcode_G28(true); }
       }
     }
 
+    void G33_cleanup(
+      #if HOTENDS > 1
+        const uint8_t old_tool_index
+      #endif
+    ) {
+      #if ENABLED(DELTA_HOME_TO_SAFE_ZONE)
+        do_blocking_move_to_z(delta_clip_start_height);
+      #endif
+      STOW_PROBE();
+      clean_up_after_endstop_or_probe_move();
+      #if HOTENDS > 1
+        tool_change(old_tool_index, 0, true);
+      #endif
+    }
+
```

Comment:
To reduce repetition, you might try…
```cpp
#if HOTENDS > 1
  #define G33_CLEANUP(T) g33_cleanup(T)
#else
  #define G33_CLEANUP(T) g33_cleanup()
#endif
```
Then all `g33_cleanup` calls can be one line…
```cpp
G33_CLEANUP(old_tool_index);
```